packages/amplify-frontend-javascript/lib/frontend-config-creator.js

```diff
@@ -289,13 +289,38 @@ function getCognitoConfig(cognitoResources, projectRegion) {
     federationTarget = 'COGNITO_IDENTITY_POOLS';
   }
 
+  let authData = {};
+
+  if (cognitoResource.frontEndConfig) {
+    if (cognitoResource.frontEndConfig.loginMechanism) {
+      authData.aws_cognito_login_mechanism = cognitoResource.frontEndConfig.loginMechanism;
```

Comment:
assuming assigning to undefined is the same as not assigning at all, all of these if statements can be removed and replaced with:
```suggestion
authData.aws_cognito_login_mechanism = cognitoResource?.frontendConfig?.logicMechanism;
```